tests/Language/AbstractTranslationTestCase.php

```diff
@@ -0,0 +1,413 @@
+            $this->foundSets($locale),
+            $this->expectedSets()
+        );
+
+        sort($filesNotConfigured);
+        $count = count($filesNotConfigured);
+
+        $this->assertEmpty($filesNotConfigured, sprintf(
+            'Failed asserting that translated language %s "%s" in "%s" locale %s configured in the main repository.',
+            $count > 1 ? 'files' : 'file',
+            implode('", "', $filesNotConfigured),
+            $locale,
+            $count > 1 ? 'are' : 'is'
+        ));
+    }
+
+    /**
+     * This tests that all language keys defined by a language file in the main CI4
+     * repository have corresponding keys in the current locale.
+     *
+     * @dataProvider localesProvider
+     */
+    final public function testAllConfiguredLanguageKeysAreIncluded(string $locale): void
+    {
+        $keysNotIncluded = [];
+
+        foreach ($this->foundSets($locale) as $file) {
+            $missing = array_diff_key(
+                $this->loadFile($file),
+                $this->loadFile($file, $locale)
+            );
+
+            foreach (array_keys($missing) as $key) {
+                $keysNotIncluded[] = substr($file, 0, -4) . '.' . $key;
+            }
+        }
+
+        sort($keysNotIncluded);
+        $count = count($keysNotIncluded);
+
+        $this->assertEmpty($keysNotIncluded, sprintf(
+            'Failed asserting that the language %s "%s" in the main repository %s included for translation in "%s" locale.',
+            $count > 1 ? 'keys' : 'key',
+            implode('", "', $keysNotIncluded),
+            $count > 1 ? 'are' : 'is',
+            $locale
+        ));
+    }
+
+    /**
+     * This tests that all included language keys in a language file for the current
+     * locale have corresponding keys in the main CI4 repository.
+     *
+     * @dataProvider localesProvider
+     */
+    final public function testAllIncludedLanguageKeysAreConfigured(string $locale): void
+    {
+        $keysNotConfigured = [];
+
+        foreach ($this->foundSets($locale) as $file) {
+            $extra = array_diff_key(
+                $this->loadFile($file, $locale),
+                $this->loadFile($file)
+            );
+
+            foreach (array_keys($extra) as $key) {
+                $keysNotConfigured[] = substr($file, 0, -4) . '.' . $key;
+            }
+        }
+
+        sort($keysNotConfigured);
+        $count = count($keysNotConfigured);
+
+        $this->assertEmpty($keysNotConfigured, sprintf(
+            'Failed asserting that the translated language %s "%s" in "%s" locale %s configured in the main repository.',
+            $count > 1 ? 'keys' : 'key',
+            implode('", "', $keysNotConfigured),
+            $locale,
+            $count > 1 ? 'are' : 'is'
+        ));
+    }
+
+    /**
+     * This tests that all included language keys in a language file for the current
+     * locale that have corresponding keys in the main CI4 repository are really translated
```

Comment:
Just to be a bit nit-picky, but all of the comments in the file says it tests that they match the CI4 repository, which isn't what's being tested here. Should specify the Shield repo.

Reply:
I think it it better that the file is in other repository like `codeigniter4/translation-tests`.
So we can share it.